### DISCUSSION TITLE
API Query

### DISCUSSION THREAD

**Discussion by saaramitay** [community/community#54859] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

Hi,

I would like you to add an API call to query in each repo the people with admin permissions on the repo and the API will contain the admin email address.

For now the only place that contains a centralized list of people in the repo with admin permissions and their email addresses is in:
https://github.com/{OrganizationName}/{RepositoryName}/people

But this does not have an API call.

Thanks!

**Comment by saifulislammz** [comment]
### ### Fetching Admin Users with Emails via GitHub API

Since GitHub does not provide a single API call for this, you can use a combination of the following API endpoints:

1. **Get Collaborators with Admin Access**
This API retrieves all collaborators in a repository along with their permissions.
Endpoint:  `GET /repos/{owner}/{repo}/collaborators`

Example Request: 
```
curl -H "Authorization: token YOUR_GITHUB_TOKEN" \
     -H "Accept: application/vnd.github.v3+json" \
     https://api.github.com/repos/{owner}/{repo}/collaborators
```

2. **Filter Out Admins**
The response contains a `permissions` field. Filter users where `"admin": true`.

3. **Get Admins’ Email Addresses**

- If a user has made their email public, you can fetch it using:
`GET /users/{username}`

- Example Request:
```
curl -H "Authorization: token YOUR_GITHUB_TOKEN" \
     -H "Accept: application/vnd.github.v3+json" \
     https://api.github.com/users/{username}
```

- This will return an email if it is public. If not, email retrieval requires admin access to GitHub Enterprise or internal methods.


**Limitations & Considerations**

1. **Email Privacy:** If a user has hidden their email, you won't get it via the public API.
2. **GitHub Enterprise:** Some enterprise versions allow fetching email addresses if you have the right permissions.
3. **Rate Limits:** GitHub API has rate limits; using authentication increases your quota.
